### PR TITLE
feat: add searchbar to collectible tabs

### LINKED ISSUE #3444: [Subtask]: E12.M2 - Feature: search bar
### Task description

Add searchbar component with the filtering feature to the collectibles gallery.
Search by name.

### Requirements

- [ ] search bar component
- [ ] search bar functionality

### Acceptance criteria

Meets the above requirements

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary

This PR adds a togglable Searchbar on the Collectible Tabs, to search the NFTs by their name

## Changelog

```
- adds a togglable Searchbar on the Collectible Tabs
- implement search logic by name
```

## Relevant Issues

Closes #3444 

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/nfts/stores/selected-account-nfts.store.ts [hunk 1]
```diff
@@ -15,3 +15,18 @@ export const selectedAccountNfts: Readable<INft[]> = derived(
...
+export const queriedNfts: Readable<INft[]> = derived(
+    [selectedAccountNfts, nftSearchTerm],
+    ([$selectedAccountNfts, $nftSearchTerm]) => {
+        let nftList = $selectedAccountNfts
+
+        if ($nftSearchTerm) {
+            nftList = nftList.filter((nft) => nft.name.toLowerCase().includes($nftSearchTerm.toLowerCase()))
+        }
+
+        return nftList.sort((nft1, nft2) => (nft2.name < nft1.name ? 1 : -1))
```
Comment (kraftjs) [inline packages/shared/lib/core/nfts/stores/selected-account-nfts.store.ts:30]: This doesn't sort alphabetically; it's sorted by unicode value. "NFT" is placed before "application". It's not an issue but we should figure out how we want to default sort the NFTs.

File: packages/desktop/views/dashboard/collectibles/views/GalleryView.svelte [hunk 2]
```diff
@@ -1,23 +1,63 @@
...
     }
 </script>
 
-<div class="flex flex-row text-left space-x-1 items-center">
-    <Text fontSize="text-14" fontWeight={FontWeight.semibold}>{localize('views.collectibles.gallery.title')}</Text>
-    <Text fontSize="text-14" fontWeight={FontWeight.semibold} color="gray-500">• {$selectedAccountNfts.length}</Text>
+<div class="flex flex-row justify-between">
+    <div class="flex flex-row text-left space-x-1 items-center">
+        <Text fontSize="text-14" fontWeight={FontWeight.semibold}>{localize('views.collectibles.gallery.title')}</Text>
+        <Text fontSize="text-14" fontWeight={FontWeight.semibold} color="gray-500">• {$selectedAccountNfts.length}</Text
```
Comment (kraftjs) [inline packages/desktop/views/dashboard/collectibles/views/GalleryView.svelte:28]: Do we want to show `$selectedAccountNfts.length` or do we want to show the number of returned NFTs from the search, so `$queriedNfts.length`?

**Review by kraftjs — COMMENTED** [review]
Impressive implementation. Was nice to review your code. Just have 1.5 comments for you to address and then it's ready for approval 😃

File: packages/shared/lib/core/nfts/stores/selected-account-nfts.store.ts (hunk 1 above)
Comment (MarkNerdi) [reply packages/shared/lib/core/nfts/stores/selected-account-nfts.store.ts:30]: ah thats true, i'll change this to `nft2.name.toLowercase() < nft1.name.toLowercase()`

**Review by kraftjs — APPROVED** [review]
Approved now, thanks for addressing comments 🔥

**Review by kraftjs — APPROVED** [review]

**Pull request merged**